packages/ai-core/src/browser/frontend-language-model-registry.ts

```diff
@@ -286,6 +292,14 @@ export class FrontendLanguageModelRegistryImpl
         throw new Error(`Could not find a tool for ${toolId}!`);
     }
 
+    error(id: string, error: Error): void {
+        if (!this.streams.has(id)) {
+            throw new Error('Somehow we got a callback for a non existing stream!');
+        }
+        const streamState = this.streams.get(id)!;
+        streamState.reject?.(error);
+    }
```

Comment:
In the `send` callback which is also called from the backend we construct the stream state if it does not exist yet. I would suggest to do the same here.

If the first feedback of a stream is an error it could happen that the error arrives here first before the corresponding stream state was created as we don't control the async flows in the backend. We don't want to throw an error in that case but just handle it like all the other cases too.